### PR TITLE
[ML] Prefer smaller models with similar performance

### PR DESCRIPTION
**Pull request by valeriy42**
For regression and classification, during hyperparameter optimization we prefer smaller models if the loss functions are otherwise comparable.

To this end, we add  0.01 * "forest number nodes" * E[GP] / "average forest number nodes" as an additional penalty.

### CODE REVIEW

**Comment by valeriy42** [comment]
@wwang500 just a heads-up: in some cases this may lead to a regression of the results by <1% (in terms of MSE, MSLE, Huber) or less (for classification).

**Review by tveasey — APPROVED** [review]
LGTM
